ppl/src/main/java/org/opensearch/sql/ppl/PPLService.java

```diff
@@ -57,6 +63,8 @@ public class PPLService {
    */
   public void execute(PPLQueryRequest request, ResponseListener<QueryResponse> listener) {
     try {
+      ExecutionEngine executionEngine =
+          catalogService.getExecutionEngine(connector).orElse(openSearchExecutionEngine);
```

Comment:
Should catalogService decide which executionEngine should be used? even for default one?

Reply:
I have implemented the same initially...but the default one is using node client coming from the upstream in the api call. 

This is not available when catalog service initializes during server start up.